lib/api/2.0/notification.js

```diff
@@ -30,7 +36,124 @@ var notificationProgressPost = controller(function(req, res){
+                         bmcCatalog = data;
+                        //Format the amqp message by adopting the rackHD notification standard
+                        amqpMessage.type = "node";
+                        amqpMessage.action = "alerts";
+                        amqpMessage.typeId = bmcCatalog.node;
+
+                        var httpReq = _.defaults(req.swagger.query || {},
+                            req.query || {},
+                            req.body || {});
+                        if(Object.keys(httpReq).length === 1 ) {
+                            var key = Object.keys(httpReq)[0];
+                            alert = JSON.parse(key);
+                            amqpMessage.data = alert;
+                            amqpMessage.data.sourceIpAddress = ip;
+                            return waterline.catalogs.findOne({source: "ipmi-fru"})
+                                .then(function(catalogIpmiFru){
+                                    if(catalogIpmiFru !== undefined &&
+                                        catalogIpmiFru.data
+                                            .hasOwnProperty("Builtin FRU Device (ID 0)")){
+                                        deviceInfoCatalog =
+                                            catalogIpmiFru.data["Builtin FRU Device (ID 0)"];
+
+                                        //Add useful information to the alert message
+                                        amqpMessage.data.rackhdId = bmcCatalog.node;
+                                        amqpMessage.data.macAddress = bmcCatalog["MAC Address"];
+                                        amqpMessage.severity = alert.Severity.toLowerCase();
+                                        amqpMessage.data.ChassisName =
+                                            deviceInfoCatalog["Board Product"];
+                                        amqpMessage.data.ServiceTag =
+                                            deviceInfoCatalog["Product Serial"];
+                                        amqpMessage.data.SN = deviceInfoCatalog["Board Serial"];
+
+                                        return notificationApiService.
+                                        alertNotification(amqpMessage);
+                                    }else{
+                                        //publish with the additional information
+                                        return notificationApiService.
+                                        alertNotification(amqpMessage);
+                                    }
+                                });
+                        }else{
+                            //Unexpected decoded "x-www-forum-urlencoded" data
+                            throw new Errors.BadRequestError(
+                                '"Unexpected decoded "x-www-forum-urlencoded" data"'
+                            );
+                        }
+                    }else{
```

Comment:
I even can't easily find which is the matching `if`, for better readability, please split this function into some smaller ones. Or you could put the error handling in advance to reduce the code nesting level. 

For example:
```javascript
if (cond) {
    doSomething();
} else {
    throw new Error();
}
```

it could be refactored into:
```javascript
if (!cond) {
    throw new Error();
}
doSomething();
``` 

You could see the function `doSomething()` has  been increased the nesting level to  the same with `if`.